src/ruby/lib/grpc/generic/client_stub.rb

```diff
@@ -149,16 +153,29 @@ def request_response(method, req, marshal, unmarshal,
                           deadline: deadline,
                           parent: parent,
                           credentials: credentials)
-      return c.request_response(req, metadata: metadata) unless return_op
-
-      # return the operation view of the active_call; define #execute as a
-      # new method for this instance that invokes #request_response.
-      c.merge_metadata_to_send(metadata)
-      op = c.operation
-      op.define_singleton_method(:execute) do
-        c.request_response(req, metadata: metadata)
+      interception_context = InterceptionContext.new(interceptors.all)
+      intercept_args = {
+        method: method,
+        request: req,
+        call: c,
+        metadata: metadata
```

Comment:
One thing on this (and other request types) was how much information we wanted to pass to interceptors. For example, do we care about passing the deadline, parent, and credentials options? Should that be encapsulated in a `call_options` hash that gets passed to the Interception Context? Looking for thoughts here.